### DISCUSSION TITLE
[FR] Add a tag filtering feature when importing annotations from Zotero

### DISCUSSION THREAD

**Discussion by wwjCMP** [aidenlx/obsidian-zotero#221] [Help]
I make a lot of annotations when reading PDFs using Zotero, and a significant portion of them are for assisting in my reading and organizing my thoughts. However, the truly important content that needs to be cited is a small part. If I were to import all the annotations into Obsidian, it would generate a lot of unnecessary data. I hope that an optional feature can be added to import annotations based on the tags in them, so only the annotations containing specific tags are imported. This way, users can have more control over which truly important annotations are imported.

thk

**Comment by asan-emirsaleh** [comment]
There is another approach: you can filter references in Zotero and select those you prefer to be exported, then export. It is more issue with Zotero: I mark pdf with annotations and highlights, but don't want to leave out-of-context notes on the Notes pane. To know what references are processed I need to do an additional step - I add emoji tag. Recently Zotero has introduced sorting by presense/absence of notes but I process papers with annotations so I need to routinely add tag for tracing annotated references.

**Comment by aidenlx** [comment]
This is doable in the template in v2, without needing a dedicated setting. Annotations carry their tags, and the content template is an ordinary loop, so you can import only the ones you tagged:

```liquid
{% for annotation in zt.annotations %}
{% assign names = annotation.tags | map: "name" %}
{% if names contains "cite" %}{% render "annotation" with annotation as zt %}{% endif %}
{% endfor %}
```

Because the managed region is re-rendered from current Zotero data on every update, retagging an annotation in Zotero is enough to pull it in or drop it on the next update.

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
